cmd/soroban-rpc/internal/test/simulate_transaction_test.go

```diff
@@ -430,7 +429,7 @@ func TestSimulateInvokeContractTransactionSucceeds(t *testing.T) {
 	ro1 := obtainedFootprint.ReadOnly[1]
 	assert.Equal(t, xdr.LedgerEntryTypeContractData, ro1.Type)
 	assert.Equal(t, xdr.ScAddressTypeScAddressTypeContract, ro1.ContractData.Contract.Type)
-	assert.Equal(t, xdr.Hash(contractID), ro1.ContractData.Contract.ContractId)
+	assert.Equal(t, xdr.Hash(contractID), *ro1.ContractData.Contract.ContractId)
```

Comment:
if testing a value in a *uint, I think that adding
```golang
require.NotNil(t, ro1.ContractData.Contract.ContractId)
```
could be a good idea - i.e. we don't want the test to panic due to segmentation fault. We want a "clean" failure.